### PR TITLE
Fix/serializing hash

### PR DESCRIPTION
**Pull request by svechinsky**
Just fixing https://github.com/mimblewimble/grin/issues/2351

### CODE REVIEW

**Comment by ignopeverell** [comment]
This fixes the issue but doesn't consider backward compatibility. As-is it's going to make syncing on floonet very difficult until everyone updates.

File: store/src/pmmr.rs [hunk 1]
```diff
@@ -193,7 +193,13 @@ impl<T: PMMRable> PMMRBackend<T> {
 				data_dir.join(PMMR_LEAF_FILE).to_str().unwrap(),
 				header.hash()
 			);
-			LeafSet::copy_snapshot(&leaf_set_path, &PathBuf::from(leaf_snapshot_path))?;
+			// Check for a ... (3 dot) ending version of the file - could probably be removed after mainnet
+			let compaitble_snapshot_path = PathBuf::from(leaf_snapshot_path.clone() + "...");
```
Comment (ignopeverell) [inline store/src/pmmr.rs]: Small typo here: compatible. To your defense, I have a hard time typing too...

Comment (svechinsky) [reply store/src/pmmr.rs]: The pitfalls of relying on autocorrect too much

**Comment by ignopeverell** [comment]
Tested fast sync a few times and it behaves as expected. Note that sync against pre-0.5.2 nodes will still very likely fail (different hash length in leaf file), but hopefully people will all be updating soon.

**Pull request merged**